rt_gene/src/rt_gene/extract_landmarks_method.py

```diff
@@ -411,39 +414,38 @@ def callback(self, color_msg):
                 pass
             else:
                 raise e
-     
-    def __detect_landmarks_one_box(self, facebox, color_img):
-        try:
-            _bb = map(int, facebox)
-            face_img = color_img[_bb[1]: _bb[3], _bb[0]: _bb[2]]
-            marks_orig = np.array(self.__detect_facial_landmarks(color_img, facebox)[0])
-
-            eye_indices = np.array([36, 39, 42, 45])
-
-            transformed_landmarks = marks_orig[eye_indices]
-            transformed_landmarks[:, 0] -= facebox[0]
-            transformed_landmarks[:, 1] -= facebox[1]
-
-            return face_img, transformed_landmarks, marks_orig
-        except Exception as e:
-            print("*** Exception in detecting landmarks from facebox ***", e)
-            return None, None, None
+    
+    @staticmethod
+    def crop_face_from_image(color_img, box):
+        _bb = map(int, box)
+        return color_img[_bb[1]: _bb[3], _bb[0]: _bb[2]]
 
-    def __detect_facial_landmarks(self, color_img, facebox):
-        marks = self.facial_landmark_nn.get_landmarks(color_img, detected_faces=[facebox])
-        return marks
+    @staticmethod
+    def transform_landmarks(landmarks, box):
+        eye_indices = np.array([36, 39, 42, 45])
+        transformed_landmarks = landmarks[eye_indices]
+        transformed_landmarks[:, 0] -= box[0]
+        transformed_landmarks[:, 1] -= box[1]
+        return transformed_landmarks
 
     def detect_landmarks(self, color_img, timestamp):
         faceboxes = self.get_face_bb(color_img)
+        if not faceboxes:
+            # reset the tracker
+            self.subject_tracker.clear_elements()
+            return
 
-        # track the new faceboxes according to the previous ones
-        self.subject_tracker.track([TrackedSubject(np.array(new_fb)) for new_fb in faceboxes])
+        landmarks = self.facial_landmark_nn.get_landmarks(color_img, detected_faces=faceboxes)
+        face_images = [LandmarkMethod.crop_face_from_image(color_img, b) for b in faceboxes]
 
-        for subject in self.subject_tracker.get_tracked_elements().values():
-            face, landmarks, marks = self.__detect_landmarks_one_box(subject.box, color_img)
-            subject.face_color = face
-            subject.transformed_landmarks = landmarks
-            subject.marks = marks
+        tracked_subjects = []
+        for b, l, f in zip(faceboxes, landmarks, face_images):
```

Comment:
Can we use more descriptive variable names than `b`, `l` and `f`?

Reply:
Done! :)